docs/sources/troubleshoot/profile.md

```diff
@@ -0,0 +1,101 @@
+---
+canonical: https://grafana.com/docs/alloy/latest/troubleshoot/profile/
+description: Learn how to profile resource consumption
+title: Profile Grafana Alloy resource consumption
+menuTitle: Profile resource consumption
+weight: 300
+---
+
+# Profile {{% param "FULL_PRODUCT_NAME" %}} resource consumption
+
+{{< param "PRODUCT_NAME" >}} is written in the Go programming language, which offers [built-in][pprof-pkg] support for profiling.
+Like other applications written in Go, you can profile {{< param "PRODUCT_NAME" >}} by sending an HTTP request, which returns a pprof Go profile.
+
+Once you have the pprof file, visualize it as a [flame graph][flame-graph] in [Grafana Pyroscope][pyroscope-getstarted].
+Alternatively, you could visualize it on [Grafana Play][pyroscope-adhoc] or locally by using Go's built-in [pprof tool][go-pprof].
+
+{{< admonition type="note" >}}
+A profile may contain sensitive information about your environment.
+You may not want to upload your profiles to a public location.
+{{< /admonition >}}
+
+The port you use to send the HTTP request is governed by Alloy's `--server.http.listen-addr` [command line argument][cmd-cli].
```

Comment:
```suggestion
The port you use to send the HTTP request is controlled by the `--server.http.listen-addr` [command line argument][cmd-cli].
```

Missed this one in the first pass.  I think we can drop the `Alloy's` and assume the reader knows we are talking about the CLI arg in Alloy here.